### DISCUSSION TITLE
Am I right that `numCalls` won't be needed once issue #585 is implemented?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [lucas-manuel/invariant-example#6] [Q&A] [answered]
I noticed that this example uses a [`numCalls`](https://github.com/lucas-manuel/invariant-examples/blob/d6b4f06d7ebe0bbaebcfd666fb00433d30ab28da/test/handlers/LpHandler.sol#L23) mapping to count the number of calls made to handler contracts.

I saw the note you left at the end of the README (that `invariant_call_summary` is for illustrative purposes only), however I just want to double-check that my understanding of [issue #585](https://github.com/foundry-rs/foundry/issues/585) in Foundry is correct.

Once that issue is implemented, there will be no need for hacks like `numCalls`, will there?

**Comment by lucas-manuel** [comment, marked as answer]
I think you might be thinking about this issue here: https://github.com/foundry-rs/foundry/issues/3607 but yeah once this is implementated the `invariant_call_summary` hacks not needed.

**Reply by PaulRBerg** [reply]
You're right - I was referring to this other issue. Thanks.
